code/modules/mob/living/silicon/robot/gripper.dm

```diff
@@ -11,39 +11,35 @@
 
 	flags = NOBLUDGEON
 
+	 //This is a list of items that can't be held even if their parent is whitelisted.
+	var/list/blacklist = list(
+		/obj/item/gun,
+		/obj/item/projectile,
+		/obj/item/storage,
+		/obj/item/modular_computer,
+		/obj/item/remains,
+		/obj/item/wheelchair,
+		/obj/item/stool,
+		/obj/item/shield
+		)
 	//Has a list of items that it can hold.
 	var/list/can_hold = list(
-		/obj/item/extinguisher,
-		/obj/item/cell,
-		/obj/item/firealarm_electronics,
-		/obj/item/airalarm_electronics,
-		/obj/item/airlock_electronics,
-		/obj/item/tracker_electronics,
-		/obj/item/stock_parts,
-		/obj/item/frame,
-		/obj/item/camera_assembly,
-		/obj/item/tank,
-		/obj/item/circuitboard,
-		/obj/item/device/assembly,//Primarily for making improved cameras, but opens many possibilities
-		/obj/item/computer_hardware,
-		/obj/item/tool_upgrade,
-		/obj/item/am_containment,
-		/obj/item/am_shielding_container,
-		/obj/item/stack/tile, //Repair floors yay
-		/obj/item/oddity/blackbox_nanoforge
-		)
+		/obj/item
+		) // Look, look with your special eyes!
 
 	var/obj/item/wrapped = null // Item currently being held.
 
+	var/list/valid_containers = list(/obj/item/storage)
+
 	var/force_holder = null //
 	var/justdropped = 0//When set to 1, the gripper has just dropped its item, and should not attempt to trigger anything
 
-/obj/item/gripper/examine(var/mob/user)
-	..()
-	if (wrapped)
-		to_chat(user, span("notice", "It is holding \the [wrapped]"))
+/obj/item/gripper/examine(mob/user)
+	if(wrapped)
+		wrapped.examine(user)
```

Comment:
If you are holding something and examine your gripper, you will examine the ITEM BEING HELD, instead.